Introduce QNX CI job

LittleHuba opened the pull request.

LittleHuba committed:
Introduce QNX CI job

PiotrKorkus committed:
cicd: add release workflow

execute tests and store artifacts

LittleHuba committed:
Add configuration for QNX on arm64

LittleHuba committed:
Test build for QNX8 on both architectures

LittleHuba committed:
Do not fail fast in matrix builds

For our current matrix builds we are inerested in the outcome of every branch independent of the other branches.

LittleHuba committed:
Bump Clang to 19.1.0

LittleHuba committed:
Disable Rust build also on QNX8 arm64

Build fails because the rust toolchain is not hermetic.

LittleHuba committed:
Clean runners to get more disk space

pawelrutkaq commented on .github/workflows/build_and_test_qnx.yml:
```diff
@@ -0,0 +1,80 @@
+# Workflow configuration for S-CORE CI - Bazel Build & Test communication module for target platforms
+# This workflow runs Bazel build and test for QNX when triggered by specific pull request events.
+
+name: Bazel Build & Test communication module (target-platforms)
```
should this use reusable QNX8 workflow from https://github.com/eclipse-score/cicd-workflows ?

LittleHuba committed:
Bump baselibs

LittleHuba committed:
Add TODO for tests on QNX

LittleHuba committed:
Adapt release workflow to use module infrastructure

LittleHuba committed:
Rename Release Verification Workflow to Coverage Report Workflow

We do not want to duplicate the logic of the workflows.
Since we have the logic except for coverage reports already as part
of other workflows we reduce this workflow to only create a coverage report.

All the other workflows we enable to also be triggered by a release creation.
